src/compiler/factoryPublic.ts

```diff
@@ -2270,16 +2270,13 @@ namespace ts {
             : node;
     }
 
-    export function createDynamicImport(argumentsArray: readonly Expression[], typeArguments: readonly TypeNode[] = []) {
-        const importToken: any = createToken(SyntaxKind.ImportKeyword);
-        return createCall(importToken, typeArguments, argumentsArray);
+    export function createDynamicImportExpression(importKeyword: Token<SyntaxKind.ImportKeyword> | undefined, argumentsArray: readonly Expression[], typeArguments: readonly TypeNode[] = []) {

```

Comment:
`typeArguments` should not have a default value. That would indicate a call like `foo<>()` (with a present yet empty type argument list) which is illegal.

Also, all arguments should be in the same order they appear in the syntax. This is because our visitor pattern often keeps track of state as we descend through the tree during transformations, and if the arguments are out of order we may end up visiting them incorrectly.

```suggestion
    export function createDynamicImportExpression(importKeyword: Token<SyntaxKind.ImportKeyword> | undefined, typeArguments: readonly TypeNode[] | undefined, argumentsArray: readonly Expression[]) {
```